### PR TITLE
Address explainer feedback from TAG

### LINKED ISSUE #146: Explainer update per TAG review feedback
I volunteered to address the TAG review feedback that concerns the [explainer](https://github.com/webmachinelearning/webnn/blob/master/explainer.md):

https://github.com/w3ctag/design-reviews/issues/570#issuecomment-768151707
https://github.com/w3ctag/design-reviews/issues/570#issuecomment-768861645

I also managed to persuade @dontcallmedom to join me in this effort 😄

### PR DESCRIPTION
**Pull request by anssiko**
Fix #146

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -144,17 +162,31 @@ Try the live version of the [WebNN NSNet2 example](https://webmachinelearning.gi
...
 
-A model-loader API can also be built atop a neural network API without losing the appeal in its simplicity. Our view is that the two APIs are complementary and not mutually exclusive to each other; however we must start with the neural network API to ensure cross-platform interoperability, a cornerstone of the web platform.
+A Model Loader API can also be built atop a neural network API without losing the appeal in its simplicity. Our view is that the two APIs are complementary and not mutually exclusive to each other; however we must start with the neural network API to ensure cross-platform interoperability, a cornerstone of the web platform.
```
Comment (huningxin) [inline explainer.md:173]:
```suggestion
A Model Loader API can also be built atop a Web Neural Network API without losing the appeal in its simplicity. Our view is that the two APIs are complementary and not mutually exclusive to each other; however we must start with the Web Neural Network API to ensure cross-platform interoperability, a cornerstone of the web platform.
```

**Review by huningxin — APPROVED** [review]
Thanks @anssiko . Looks good with a nit. Please take a look.

**Comment by anssiko** [comment]
@huningxin thanks!

**Review by wchao1115 — APPROVED** [review]

**Comment by anssiko** [comment]
Thanks @wchao1115!

@dontcallmedom we'll discuss the explainer on our [4 March 2021 call](https://github.com/webmachinelearning/meetings/blob/master/telcons/2021-03-04-agenda.md). In addition to the changes in this PR, we'd welcome your general feedback on how to balance the informative content currently split across the explainer and the spec (e.g. introduction, use cases, example code).

The explainer also links to interactive [WebNN Code Editor](https://webmachinelearning.github.io/webnn-samples/code/?example=mul_add.js) where the reader can run the examples live in the browser using the polyfill. Any thoughts on linking such interactive content from the spec itself?

File: explainer.md [hunk 2]
```diff
@@ -2,6 +2,24 @@
...
+- Skeleton detection
+- Style transfer
+- Super resolution
+- Image captioning
+- Machine translation
+- Noise suppression
+
+These use cases and more are elaborated in the [use cases section of the API specification](https://webmachinelearning.github.io/webnn/#usecases).
+
+While some of these use cases can be implemented in a constrained manner with existing Web APIs (e.g. WebGL graphics API, as demonstrated by the [WebNN API polyfill](https://github.com/webmachinelearning/webnn-polyfill) and the first-wave JS ML frameworks), no access to platform capabilities beneficial for ML such as dedicated ML hardware accelerators constraints the scope of experiences and leads to inefficient implementations on modern hardware. This disadvantages the web platform in comparison to native platforms.
```
Comment (dontcallmedom) [inline explainer.md:19]:
```suggestion
While some of these use cases can be implemented in a constrained manner with existing Web APIs (e.g. WebGL graphics API, as demonstrated by the [WebNN API polyfill](https://github.com/webmachinelearning/webnn-polyfill) and the first-wave JS ML frameworks), the lack of access to platform capabilities beneficial for ML such as dedicated ML hardware accelerators constraints the scope of experiences and leads to inefficient implementations on modern hardware. This disadvantages the web platform in comparison to native platforms.
```

File: explainer.md [hunk 3]
```diff
@@ -2,6 +2,24 @@
 
 ## What’s all this then?
 
+[Machine Learning](https://en.wikipedia.org/wiki/Machine_learning), and in particular its subfield [Deep Learning](https://en.wikipedia.org/wiki/Deep_learning) with its various neural network architectures, enable new compelling user experiences for web applications. Use cases range from intelligent video conferencing to privacy-enhancing and accessibility-improving features. Some common use cases that apply to a wide range of web applications include e.g.:
```
Comment (dontcallmedom) [inline explainer.md:5]:
```suggestion
[Machine Learning](https://en.wikipedia.org/wiki/Machine_learning), and in particular its subfield [Deep Learning](https://en.wikipedia.org/wiki/Deep_learning) with its various neural network architectures, enables new compelling user experiences for web applications. Use cases range from improved video conferencing to accessibility-improving features, with potential improved privacy over cloud-based solutions. Some common use cases that apply to a wide range of web applications include e.g.:
```

**Comment by dontcallmedom** [comment]
made a couple of comments on the PR which otherwise looks good to me; I noticed the explainer uses AI & ML - I think we should only reference AI once (e.g. at the very start) and then stick to ML which is a better fit for this spec.

With regard to the live code editor, I would keep it in the explainer / readme; I don't particular object to having in the spec too, but specs are more targeted at implementers, and it's not obvious to me that such a link would directly help that audience.

**Comment by anssiko** [comment]
Thanks @dontcallmedom!

@wchao1115 I think the AI & ML usage comment touched content you contributed, so please check that my rename https://github.com/webmachinelearning/webnn/pull/148/commits/46ab045f2527b4aa5a027eaee5854e463753aa10 did not violate your artistic expression :-)

**Comment by anssiko** [comment]
We discussed this PR on [WebML CG Teleconference – 4 March 2021](https://www.w3.org/2021/03/04-webmachinelearning-minutes.html#t04) and no additional comments or concerns were raised, so I'll merge this PR.

Thanks @dontcallmedom for suggestions and @wchao1115 and @huningxin for the review.

Couple of notes regarding the relationship of the spec and the explainer:

- This explainer is considered a living document, to be maintained and kept in sync with the API spec as it evolves.
- We may move some informative text between the spec and the explainer as we progress, but should avoid unnecessary duplication.
- The general guidance is to keep the spec focused on the implementer audience, and the explainer catering for web developers and other groups not familiar with this domain.

**Pull request merged**